`test_max_pooling_nd` is flaky

This is part of https://github.com/chainer/chainer/issues/6903 effort.

- Failure example: https://ci.preferred.jp/chainer.py37/2267/
- Failure test case: `TestMaxPoolingND_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_13_{cover_all=False, dtype=float16, in_dims=(4, 3)}.test_backward`
- Tolerance:
```
E           Not equal to tolerance rtol=0.01, atol=0.001
E           
E           Mismatch: 100%
E           Max absolute difference: 0.00464076
E           Max relative difference: 0.01566845
```

--- a/tests/chainer_tests/functions_tests/pooling_tests/test_max_pooling_nd.py
+++ b/tests/chainer_tests/functions_tests/pooling_tests/test_max_pooling_nd.py
@@ -40,8 +40,6 @@ from chainer_tests.functions_tests.pooling_tests import pooling_nd_helper
     ])
 class TestMaxPoolingND(testing.FunctionTestCase):
 
-    dodge_nondifferentiable = True
-
     def setUp(self):
         self.ndim = len(self.in_dims)
         self.ksize = (3,) * self.ndim
@@ -56,7 +54,12 @@ class TestMaxPoolingND(testing.FunctionTestCase):
 
     def generate_inputs(self):
         x_shape = (2, 3) + self.in_dims
-        x = numpy.random.randn(*x_shape).astype(self.dtype, copy=False)
+        if self.test_name in ('test_backward', 'test_double_backward'):
+            x = numpy.arange(functools.reduce(mul, x_shape), dtype=self.dtype)
+            x = x.reshape(x_shape)
+            x = 2 * x / x.size - 1
+        else:
+            x = numpy.random.randn(*x_shape).astype(self.dtype, copy=False)
         return x,
 
     def forward(self, inputs, device):

